src/main/java/Duke/exception/InvalidInputException.java

```diff
@@ -0,0 +1,24 @@
+package duke.exception;
+
+/**
+ * Represents an invalid input. Thrown when user input is invalid.
+ */
+public class InvalidInputException extends DukeException {
+
+    private String err;
+
+    public InvalidInputException(String str) {
+        super(str);
+        this.err = str;
```

Comment:
Should `this.err` simply be `err` as it is not shadowed by a method or constructor parameter?
```suggestion
        err = str;
```